### PR TITLE
Generate payment link for transfer

### LINKED ISSUE #1316: Generate payment link for transfer
User can input their address and put the amount in dollars

In return, users can generate payment links and share them with others by clicking button

- Add button to generate this payment link and copy to clipboard

Add it probably next to USD? 
Choose icon which best represents "creating payment link"
image

### PR DESCRIPTION
**Pull request by prachi00**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇  _ Do a quick check before the merge. 

### PR type
- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

### Before submitting Pull Request, please make sure:
- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted screenshot of demonstrated change in this PR

### Optional
- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything works
- [ ] I found edge cases

### What's new?
- [x] PR closes #1316
- [x] Changed the location, wasn't fitting well right next to the usd

### Had issue bounty label ?
- [x] Fill up your KSM address: 
[Payout](https://kodadot.xyz/transfer/?target=EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU)

### Community participation
- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot
- [x] My fix has changed **something** on UI, a screenshot for others, is best to understand changes.
Screenshot 2021-11-27 at 10 10 28 PM

New
Screenshot 2021-11-29 at 8 13 49 PM
:

### CODE REVIEW

File: src/components/transfer/Transfer.vue [hunk 1]
```diff
@@ -276,6 +291,10 @@ export default class Transfer extends Mixins(
     window.open(url, '_blank')
   }
 
+  protected generatePaymentLink(): string {
+    return `https://nft.kodadot.xyz/transfer?target=${this.destinationAddress}&usdamount=${this.usdValue}&donation=true`;
```
Comment (vikiival) [inline src/components/transfer/Transfer.vue:295]: instead of `https://nft.kodadot.xyz/` use `${window.location.origin}`

**Review by vikiival — CHANGES_REQUESTED** [review]
I would personally put the button down (next to submit?)

**Comment by prachi00** [comment]
> I would personally put the button down (next to submit?)

@vikiival done, changed position

File: src/components/transfer/Transfer.vue [hunk 2]
```diff
@@ -79,6 +82,18 @@
              >
                <b-icon size="is-small" pack="fas" icon="link" />
              </b-button>
+            <b-field class="mb-3 ml-3" v-show="destinationAddress">
```
Comment (roiLeo) [inline src/components/transfer/Transfer.vue:85]: you don't need `mb-3 ml-3`class here

**Review by roiLeo — CHANGES_REQUESTED** [review]
checked buttons on mobile view and it looks like it's missing some space.
Screenshot 2021-11-29 at 18-22-39 KodaDot - Kusama NFT Market Explorer Low Carbon NFTs

You should check [Buefy buttons ](https://buefy.org/documentation/button/) doc to replace `<b-field>`

File: src/components/transfer/Transfer.vue [hunk 3]
```diff
@@ -329,6 +348,15 @@ export default class Transfer extends Mixins(
...
     }
+
+    .paymentLink {
+      margin-left: 1rem;
+      @media screen and (max-width: 1023px) {
+        margin-left: 0rem;
+        margin-top: 1rem;
+      }
+    }
+
```
Comment (roiLeo) [inline src/components/transfer/Transfer.vue:359]:
```suggestion

```

**Review by roiLeo — CHANGES_REQUESTED** [review]
Screenshot 2021-11-30 at 09-07-09 KodaDot - Kusama NFT Market Explorer Low Carbon NFTs

You don't need magic css to fix this.

use `<div class="buttons">` as a wrapper

// TODO:
- remove line 81
- remove class paymentLink
- replace button wrapper `<b-field>` by `<div class="buttons">`

File: src/components/transfer/Transfer.vue (hunk 3 above)
Comment (yangwao) [reply src/components/transfer/Transfer.vue:359]: image

**Comment by yangwao** [comment]
Let me know whenever I can test through

**Comment by yangwao** [comment]
@prachi00 let's update the URL live in the address bar once I'll put the address and USD amount there :)

**Comment by prachi00** [comment]
> @prachi00 let's update the URL live in the address bar once I'll put the address and USD amount there :)

@yangwao @roiLeo have done the changes mentioned, you can test now, let me know if there are still any issues you see, thanks

**Review by roiLeo — APPROVED** [review]
lgtm, what will happen when adding new currency #939?

**Comment by yangwao** [comment]
> > @prachi00 let's update the URL live in the address bar once I'll put the address and USD amount there :)
> 
> @yangwao @roiLeo have done the changes mentioned, you can test now, let me know if there are still any issues you see, thanks

Seems when I open https://deploy-preview-1329--nftkodadot.netlify.app/transfer?usdamount=100&target=EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU it won't make me able to submit the transaction

**Comment by yangwao** [comment]
Payout sent https://kusama.subscan.io/extrinsic/10331792-15
@prachi00 let's pick #1361 to finish it there, merging this one, so it's not blocking at this stage and we can release other stuff :)
Looking forward in #1361

**Pull request merged**